native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
```

Comment:
Sure, check out the `lib` folder in the `typestyle` package (cc @basarat). The thing is that if you're publishing a package with multiple importable things, you should generally include the `.d.ts` files in directories adjacent to the source themselves. This is to mirror the exact path so that they can be imported the same way.

Since I'm looking at the package now, it looks like there should be a `native.d.ts`, not an `index.d.ts` within the `native/` directory.